memphis/memphis.go

```diff
@@ -28,17 +31,77 @@ type MemphisOutput struct {
 	FailedMessages []MemphisMsgWithError `json:"failed_messages"`
 }
 
-// EventHandlerFunction gets the message payload as []byte, message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// HandlerType functions get the message payload as []byte (or any), message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
 // error should be returned if the message should be considered failed and go into the dead-letter station.
 // if all returned values are nil the message will be filtered out of the station.
-type EventHandlerFunction func([]byte, map[string]string, map[string]string) ([]byte, map[string]string, error)
+type HandlerType func(any, map[string]string, map[string]string) (any, map[string]string, error)
+
+type HandlerOption func(*HandlerOptions) error
+
+type HandlerOptions struct {
+	Handler           HandlerType
+	UserObject        any
+}
+
+func ObjectOption(schema any) HandlerOption {
+	return func(handlerOptions *HandlerOptions) error {
+		handlerOptions.UserObject = schema
+		return nil
+	}
+}
+
+func UnmarshalIntoStruct(data []byte, userStruct any) error {
+	// Unmarshal JSON data into the struct
+	err := json.Unmarshal(data, userStruct)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkValidStruct(userStruct any) error {
```

Comment:
can remove this validation - it's ok to fail on the Unmarshal function if the user didn't pass the right format

Reply:
https://github.com/memphisdev/memphis-functions.go/pull/6/commits/8df5a62c5dd353cfc9ccbcadeb70eecf1e288de9